pkg/sdn/plugin/multitenant.go

```diff
@@ -146,52 +141,20 @@ func (mp *multiTenantPlugin) RefVNID(vnid uint32) {
 	}
 }
 
-func (mp *multiTenantPlugin) UnrefVNID(vnid uint32) {
+func (mp *multiTenantPlugin) RemoveVNIDRules(vnid uint32) {
 	if vnid == 0 {
 		return
 	}
 
-	mp.vnidRefsLock.Lock()
-	defer mp.vnidRefsLock.Unlock()
-	if mp.vnidRefs[vnid] == 0 {
-		glog.Warningf("refcounting error on vnid %d", vnid)
-		return
-	}
-	mp.vnidRefs[vnid] -= 1
-	if mp.vnidRefs[vnid] > 0 {
-		return
-	}
-	glog.V(5).Infof("UnrefVNID %d removing rule", vnid)
+	mp.vnidInUseLock.Lock()
+	defer mp.vnidInUseLock.Unlock()
+	mp.vnidInUse[vnid] = false
+
+	glog.V(5).Infof("RemoveVNIDRules %d", vnid)
 
 	otx := mp.node.oc.NewTransaction()
-	otx.DeleteFlows("table=80, reg0=%d, reg1=%d", vnid, vnid)
+	otx.DeleteFlows("table=80, reg1=%d", vnid)
```

Comment:
This function is supposed to be removing all flows that locally deliver traffic with a certain VNID. *Any* flow with "reg1=${vnid}" meets that definition. As it happens, multitenant doesn't add any reg1 rules that don't also check reg0, but maybe it would in the future. (I was also trying to merge multitenant:RemoveVNIDFlows() and networkpolicy:RemoveVNIDFlows() together, but that doesn't quite work because they still need to update their internal "vnid is in use" state. But maybe the OVS parts of them could be abstracted out later.)